### PR TITLE
How-to use contrail-ansible-deployer to deploy Windows nodes

### PR DESCRIPTION
**Pull request by mclapinski**

### CODE REVIEW

File: doc/contrail_deployment/contrail_deployment_manual.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+* On the Ansible machine:
+  * `git clone git@github.com:Juniper/contrail-ansible-deployer.git`
+  * `cd contrail-ansible-deployer`
+  * `vim config/instances.yaml`
+    * for Windows nodes use `bms_win` dict instead of regular `bms`
+    * roles supported on Windows are `vrouter` and `win_docker_driver`
+    * `WINDOWS_PHYSICAL_INTERFACE` should be set to dataplane interface alias
+    * specify `WINDOWS_ENABLE_TEST_SIGNING` if you want to install untrusted artifacts
+    * `WINDOWS_DEBUG_DLLS_PATH` should be set to path on Ansible machine containing MSVC 2015 debug dlls, specifically: `msvcp140d.dll`, `ucrtbased.dll` and `vcruntime140d.dll`
+    * refer to `config/instances.yaml.bms_win_example` for an example
```
Comment (aethyrxeo) [inline doc/contrail_deployment/contrail_deployment_manual.md]: IMO If we use an example, it should be for an "end case" e.g. "final" configuration of 2 compute nodes with templated(like `ip: WIN_COMPUTE_1_IP`)  ip and credentials.

File: doc/contrail_deployment/contrail_deployment_manual.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# How to deploy Contrail with Windows compute nodes
+
+## Prerequisites
+
+* Machine with CentOS 7.4 installed
```
Comment (aethyrxeo) [inline doc/contrail_deployment/contrail_deployment_manual.md]: This is not true for this case.

**Comment by aethyrxeo** [comment]
`Configure_instances` playbook gets stuck on `Start docker daemon` task, because the deployer doesn't create `/etc/docker/daemon.json` file.

**Comment by aethyrxeo** [comment]
`Install_contrail` playbook fails on `get auth token from keystone` task, when keystone credentials aren't provided.

File: doc/contrail_deployment/contrail_deployment_manual.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+* Machine(s) with Windows Server 2016 and updates installed
+* Machine for running Ansible playbooks (Linux or Windows with WSL)
+
+## Steps
+
+* On the Ansible machine:
+  * `git clone git@github.com:Juniper/contrail-ansible-deployer.git`
+  * `cd contrail-ansible-deployer`
+  * `vim config/instances.yaml`
```
[the remark this answers is not in the document]
Comment (aethyrxeo) [reply doc/contrail_deployment/contrail_deployment_manual.md]: Actually we need to include information about keystone and openstack.

**Comment by aethyrxeo** [comment]
In `wait for docker driver startup` task ansible actually doesn't try to start docker driver. It's probably a bug in win_nssm.

**Comment by aethyrxeo** [comment]
Please relate to [technical writing guide]( https://en.wikiversity.org/wiki/Technical_writing_style) especially `five rules of concise communication` 

E.g. Change `WINDOWS_PHYSICAL_INTERFACE should be set to dataplane interface alias` to `Set WINDOWS_PHYSICAL_INTERFACE to dataplane interface alias`

**Comment by aethyrxeo** [comment]
`WINDOWS_CONTAINER_REGISTRY` and `CONTAINER_REGISTRY` use the same variable `CONTRAIL_VERSION` to specify docker repo tag. This does not allow to pull repo for non-windows nodes from other tags.

File: doc/contrail_deployment/contrail_deployment_manual.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      * Roles supported on Windows are `vrouter` and `win_docker_driver`. Specify them.
+      * Set `WINDOWS_PHYSICAL_INTERFACE` to dataplane interface alias
+      * Specify `WINDOWS_ENABLE_TEST_SIGNING` if you want to install untrusted artifacts (this is needed at this moment)
+      * Set `WINDOWS_DEBUG_DLLS_PATH` to path on Ansible machine containing MSVC 2015 debug dlls, specifically: `msvcp140d.dll`, `ucrtbased.dll` and `vcruntime140d.dll`
+    * To deploy a controller for windows compute nodes:
+      * Configure as it would be a controller node for a linux ecosystem.
+      * For now docker driver on windows needs keystone set up on controller:
+        * Add openstack-* roles to the controller node and set `CLOUD_ORCHESTRATOR` to `openstack`.
+        * Fill keystone credentials and kolla config. Check `config/instances.yaml.openstack_example`
+    * Refer to `config/instances.yaml.bms_win_example` or `config/instances.yaml.bms_win_full_example` for an example
```
Comment (mclapinski) [inline doc/contrail_deployment/contrail_deployment_manual.md]: Could you explain that `bms_win_example` is for people who already have a controller and `bms_win_full_example` is for people without controller?

**Review by jablonskim — APPROVED** [review]
Looks good to me.
What do you think about adding this information: "Configure_instances playbook gets stuck on Start docker daemon task, because the deployer doesn't create /etc/docker/daemon.json file."?

**Comment by aethyrxeo** [comment]
@jablonskim Done.

File: doc/contrail_deployment/contrail_deployment_manual.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+* Machine with CentOS 7.4 installed
+* Machine(s) with Windows Server 2016 and updates installed
+* Machine for running Ansible playbooks (Linux or Windows with WSL)
+
+## Steps
+
+* On the Ansible machine:
+  * `git clone git@github.com:Juniper/contrail-ansible-deployer.git`
+  * `cd contrail-ansible-deployer`
```
Comment (pmarcinkiewicz1) [inline doc/contrail_deployment/contrail_deployment_manual.md]: Please add pip install -r python-requirements.txt

Comment (jablonskim) [reply doc/contrail_deployment/contrail_deployment_manual.md]: We will think about installing it during `configure_instances`

File: doc/contrail_deployment/contrail_deployment_manual.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  * `vim config/instances.yaml`
+    * To deploy a windows compute node:
+      * Use `bms_win` dict instead of regular `bms`
```
Comment (pmarcinkiewicz1) [inline doc/contrail_deployment/contrail_deployment_manual.md]: Consider addinginstance.yaml template with comments which line need to be changed, example:

instances:
  ctrl:
    provider: bms
    ip: 10.7.0.85  #  <- Provide controller ip address here

File: doc/contrail_deployment/contrail_deployment_manual.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      * Roles supported on Windows are `vrouter` and `win_docker_driver`. Specify them
+      * Set `WINDOWS_PHYSICAL_INTERFACE` to dataplane interface alias
+      * Specify `WINDOWS_ENABLE_TEST_SIGNING` if you want to install untrusted artifacts (this is needed at this moment)
+      * Set `WINDOWS_DEBUG_DLLS_PATH` to path on Ansible machine containing MSVC 2015 debug dlls, specifically: `msvcp140d.dll`, `ucrtbased.dll` and `vcruntime140d.dll`
+      * For now `configure_instances` playbook gets stuck on `Start docker daemon` task,
+      because docker's bridge interface has default subnet the same as LAN where contrail nodes reside. If this happens, do the following:
+        * On controller:
+          * Create file `/etc/docker/daemon.json` with `bip` variable set to IP in CIDR notation not conflicting with your LAN
```
Comment (pmarcinkiewicz1) [inline doc/contrail_deployment/contrail_deployment_manual.md]: Add template of daemon.json file here.

File: doc/contrail_deployment/contrail_deployment_manual.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      * For now docker driver on windows needs keystone set up on controller:
+        * Add openstack-* roles to the controller node and set `CLOUD_ORCHESTRATOR` to `openstack`
+        * Fill keystone credentials and kolla config. Check `config/instances.yaml.openstack_example`
+    * Refer to examples for help:
+      * `config/instances.yaml.bms_win_example` if you have already deployed controller and you only want windows compute nodes
+      * `config/instances.yaml.bms_win_full_example` if you want to deploy controller and windows compute nodes together
+  * Proceed with running Ansible playbooks:
+    * `sudo -H ansible-playbook -e orchestrator=openstack -i inventory/ playbooks/configure_instances.yml`
+    * `sudo -H ansible-playbook -i inventory playbooks/install_openstack.yml`
+    * `sudo -H ansible-playbook -e orchestrator=openstack -i inventory/ playbooks/install_contrail.yml`
```
Comment (pmarcinkiewicz1) [inline doc/contrail_deployment/contrail_deployment_manual.md]: Please add info how to check that deployed setup works.

**Pull request merged**